### PR TITLE
Sorted methods for collections and added a note.

### PR DESCRIPTION
**Pull request by thinkingmedia**
fixes cakephp/cakephp#8446

I added a note under Quick Example about using the ``collection()`` global function which 

Sorted the methods so that they appear in alphabetical order.

### CODE REVIEW

**Comment by thinkingmedia** [comment]
oops, this branch included changes from my other pull request. sorry.

File: en/core-libraries/collections.rst [hunk 1]
```diff
@@ -208,6 +224,25 @@ instances by the ORM) you may want to group results by date::
...
+
+    $new = $collection->map(function ($value, $key) {
+        return $value * 2;
+    });
+
+    // $result contains ['a' => 2, 'b' => 4, 'c' => 6];
+    $result = $new->toArray();
+
+The ``map()`` method will create a new iterator which lazily creates
+the resulting items when iterated.
```
Comment (markstory) [inline en/core-libraries/collections.rst]: Why did this move? It is an 'iterating' method for the most part.

File: en/core-libraries/collections.rst [hunk 2]
```diff
@@ -337,52 +377,23 @@ function using the ``some()`` method::
...
 Aggregation
 ===========
 
-.. php:method:: reduce(callable $c)
+.. php:method:: max(string|callable $callback, $type = SORT_NUMERIC)
 
-The counterpart of a ``map()`` operation is usually a ``reduce``. This
-function will help you build a single result out of all the elements in a
-collection::
+The same can be applied to the ``max()`` function, which will return a single
```
Comment (markstory) [inline en/core-libraries/collections.rst]: With the new ordering, I'm not sure this sentence makes sense anymore. What is 'same' here now?

File: en/core-libraries/collections.rst (hunk 1 above)
Comment (thinkingmedia) [inline en/core-libraries/collections.rst]: I'll fix that.

File: en/core-libraries/collections.rst (hunk 2 above)
Comment (thinkingmedia) [inline en/core-libraries/collections.rst]: I'll give everything a read over to make sure it makes sense.

Comment (markstory) [inline en/core-libraries/collections.rst]: Thanks :smile:

**Comment by thinkingmedia** [comment]
I will review this again and make another PR when it's ready.

**Pull request closed without merging**